Tasks/DockerV2/docker.ts

```diff
@@ -7,60 +7,64 @@ import ContainerConnection from "azure-pipelines-tasks-docker-common-v2/containe
 import { getDockerRegistryEndpointAuthenticationToken } from "azure-pipelines-tasks-docker-common-v2/registryauthenticationprovider/registryauthenticationtoken";
 
 tl.setResourcePath(path.join(__dirname, 'task.json'));
+let registryAuthenticationToken: RegistryAuthenticationToken;
```

Comment:
Curious about this reordering. Also remove blank line at line 11

Reply:
Reordering was necessary here because of the nature of the MSI authentication and the getting of the ACR token from two API calls. The SPN connection is synchronous in nature as the only two parameters needed can be received in constant time (the serviceprincipalid and the serviceprincipalkey). With MSI authentication, the first API call would get the AAD token and the second would get the ACR token. In this way, if the code was not asynchronous, we would not be able to initialize the registryAuthenticationToken variable correctly. I also noticed that the rest of the code uses the registryAuthenticationToken variable, so I decided to initialize the variable such that the token could be retrieved and then used in future transactions. This does not change the SPN login as there is nothing there that would result in the code needing to wait on the registry authentication token needing to be provided.